app/docker/components/datatables/networks-datatable/networksDatatableController.js

```diff
@@ -0,0 +1,22 @@
+angular.module('portainer.docker')
+  .controller('NetworksDatatableController', ['$scope', '$controller',
+    function ($scope, $controller) {
+      angular.extend(this, $controller('GenericDatatableController', {$scope: $scope}));
```

Comment:
I was not aware of the `angular.extend` function ! This is brilliant, actually most of the custom Datatable controllers are duplicate of the generic datatable controller with some extended code. I believe that we will be able to leverage this to refactor these custom controllers and remove code duplication.

Reply:
I'm not really familiar with angular but I just checked the docs and this seems to work great.

Reply:
@deviantony already used it [here](https://github.com/portainer/portainer/pull/2648/files#diff-363850d936d3b6b1a3bf05f03ececb58) ;) even if we can definitely apply this to the *DatatableControllers, as we introduced the ES migration process it would be a loss of time to do it now imo

Reply:
Great, yes it was just a note not something to work on at the moment